Sources/StructuredData.swift

```diff
@@ -66,180 +66,123 @@ public enum StructuredData {
     }
 
     public var isBool: Bool {
-        switch self {
-        case .boolValue: return true
-        default: return false
+        if case .boolValue = self {
+            return true
         }
+        return false
     }
 
     public var isNumber: Bool {
-        switch self {
-        case .numberValue: return true
-        default: return false
+        if case .numberValue = self {
+            return true
         }
+        return false
     }
 
     public var isString: Bool {
-        switch self {
-        case .stringValue: return true
-        default: return false
+        if case .stringValue = self {
+            return true
         }
+        return false
     }
 
     public var isBinary: Bool {
-        switch self {
-        case .binaryValue: return true
-        default: return false
+        if case .binaryValue = self {
+            return true
         }
+        return false
     }
 
     public var isArray: Bool {
-        switch self {
-        case .arrayValue: return true
-        default: return false
+        if case .arrayValue = self {
+            return true
         }
+        return false
     }
 
     public var isDictionary: Bool {
-        switch self {
-        case .dictionaryValue: return true
-        default: return false
+        if case .dictionaryValue = self {
+            return true
         }
+        return false
     }
 
     public var bool: Bool? {
-        switch self {
-        case .boolValue(let b): return b
-        default: return nil
-        }
+        return try? get()
     }
 
     public var double: Double? {
-        switch self {
-        case .numberValue(let n): return n
-        default: return nil
-        }
+        return try? get()
     }
 
     public var int: Int? {
-        if let v = double {
-            return Int(v)
-        }
-        return nil
+        return double.flatMap({Int($0)})
     }
 
     public var uint: UInt? {
-        if let v = double {
-            return UInt(v)
-        }
-        return nil
+        return double.flatMap({UInt($0)})
     }
 
     public var string: String? {
-        switch self {
-        case .stringValue(let s): return s
-        default: return nil
-        }
+        return try? get()
     }
 
     public var binary: Data? {
-        switch self {
-        case .binaryValue(let d): return d
-        default: return nil
-        }
+        return try? get()
     }
 
     public var array: [StructuredData]? {
-        switch self {
-        case .arrayValue(let array): return array
-        default: return nil
-        }
+        return try? get()
     }
 
     public var dictionary: [String: StructuredData]? {
-        switch self {
-        case .dictionaryValue(let dictionary): return dictionary
-        default: return nil
-        }
+        return try? get()
     }
 
     public func get<T>() -> T? {
```

Comment:
Yes, `try?` was leaking, and I’m not sure if that’s fixed in current snapshots. But in general, I vote for removal of this method too, it’s redundant.